rfcs/0000-dsl-driven-rda-code-generation.md

```diff
@@ -0,0 +1,681 @@
+    messageClassName: gov.cms.mpsm.rda.v1.fiss.FissClaim
+    entityClassName: gov.cms.bfd.model.rda.PreAdjFissClaim
+    transformerClassName: gov.cms.bfd.pipeline.rda.grpc.source.FissClaimTransformer2
+    table:
+      name: FissClaims
+      schema: pre_adj
+      primaryKeyColumns:
+        - dcn
+      columns:
+        - name: dcn
+          sqlType: varchar(23)
+          nullable: false
+        - name: sequenceNumber
+          sqlType: bigint
+          nullable: false
+        - name: hicNo
+          sqlType: varchar(12)
+          nullable: false
+        # ... skipping lots of columns for the sake of this example ...
+    transformations:
+      - from: dcn
+        optional: false
+      - from: seq
+        to: sequenceNumber
+        optional: false
+      - from: hicNo
+        optional: false
+      # ... skipping lots of transforms for the sake of this example ...
+    arrays:
+      - from: fissProcCodes
+        to: procCodes
+        mapping: FissProcCode
+      - from: fissDiagCodes
+        to: diagCodes
+        mapping: FissDiagnosisCode
+      - from: fissPayers
+        to: payers
+        mapping: FissPayer
+        namePrefix: "payer"
+````
+
+The example illustrates the three detail tables associated with each `FissClaim`.
+The RDA API sends these as `repeated` fields in the protobuf definition and the plugin maps them to detail entities in the JPA classes.
+Each one references the field in the protobuf message and the entity class and the mapping used to define the detail table.
+
+The classes generated by the plugin rely on a few utility classes that are defined in a separate library module.
+This library is added as a dependency in the modules that require them and is the only part of the plugin that ships with the server and/or pipeline.
+
+The plugin itself is easily triggered from each module by adding a `<plugin>` element to the `<build>` element of the module's `pom.xml` file.  For example:
+
+````XML
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>gov.cms.bfd</groupId>
+                <artifactId>bfd-model-rda-codegen-plugin</artifactId>
+                <version>${project.version}</version>
+                <configuration>
+                    <mappingFile>${mapping.directory}/mapping.yaml</mappingFile>
+                    <outputDirectory>${project.build.directory}/generated-sources/transformers</outputDirectory>
+                </configuration>
+                <executions>
+                    <execution>
+                        <goals>
+                            <goal>transformers</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+````
+
+
+
+### Proposed Solution: Detailed Design
+[Proposed Solution: Detailed Design]: #proposed-solution-detailed-design
+
+The maven plugin would have a simple structure.  Each goal would follow these steps:
```

Comment:
I think we may need to step back first and talk about our options for interpreting the DSL and what a maven plugin solution brings to the table.  Even with the section above (lines 183 to 209), I think there is a leap here that others won't grasp.

Reply:
I have expanded on this topic to make it clearer.